railties/test/isolation/abstract_unit.rb

```diff
@@ -419,10 +419,16 @@ def remove_from_file(file, str)
 
     def app_file(path, contents, mode = "w")
       file_name = "#{app_path}/#{path}"
+
+      # Lets wait a bit to allow FileUpdateChecker detects the file change.
+      sleep 1 if File.exist?(file_name)
```

Comment:
This will slow down a lot of tests only because of one. I don't think we should do that. I think it would be better to use a template per string in that test.

Reply:
It is true, it will slow down a bit some tests;  This is the "assets debugging" group: [master](https://buildkite.com/rails/rails/builds/80897#b3165bd3-e739-4053-a9bc-1f1c2e6b4e0f)(2m48s) vs [this pr](https://buildkite.com/rails/rails/builds/80884#491d55af-8dbc-455b-968f-4b31405c17cc)(3m22s).

I am ok with the [template per string](https://github.com/rails/rails/pull/43188) approach, but this PR makes "app_file" method more robust in my opinion. In any case, we can revisit this topic in the future if something similar happens again.